### PR TITLE
[PM-1722] gracefully fail if site prompts user for passkey on load

### PR DESCRIPTION
**Pull request by Jingo88**
```
- [ ] Bug fix
- [ X ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Some sites will prompt you to use your passkey the moment you navigate to their sign-in page. Our objective is to gracefully fail that request, and the user will have to take some form of action (clicking use passkey, typing in username, etc) to use their passkey. 

## Code changes

page-script has been updated to look for an `mediation` key in the `options` argument for `navigate credentials get`. It will throw an error if the mediation value is `optional`

## Testing Locally

I found two sites that will prompt for using an existing passkey on load. With the new `mediation optional` logic the users will have to take action to use their passkeys. Those sites are [Kayak](https://www.kayak.com/) and [BestBuy](https://www.bestbuy.com/)

## Screenshots

Kayak after the user submits their email

kayak_passkey

BestBuy after the user clicks "sign in with passkey"

bestbuy_passkey

### CODE REVIEW

**Review by gbubemismith — COMMENTED** [review]
Thanks for working on this. LGTM! @coroiu, would you mind reviewing this as well?

File: apps/browser/src/vault/fido2/content/page-script.ts [hunk 1]
```diff
@@ -95,6 +95,11 @@ navigator.credentials.get = async (
   abortController?: AbortController
 ): Promise<Credential> => {
   const fallbackSupported = browserNativeWebauthnSupport;
+
+  if (options?.mediation && options.mediation !== "optional") {
```
Comment (coroiu) [inline apps/browser/src/vault/fido2/content/page-script.ts:99]: suggestion: simplify:)
```suggestion
  if (options?.mediation !== "optional") {
```

File: apps/browser/src/vault/fido2/content/page-script.ts [hunk 2]
```diff
@@ -95,6 +95,11 @@ navigator.credentials.get = async (
...
+    throw new Error("Webauthn Conditional UI not supported in this browser.");
```
Comment (coroiu) [inline apps/browser/src/vault/fido2/content/page-script.ts:100]: issue: I'm not sure that we wanna short circuit the request like this. The browser might still have native support for conditional UI and throwing an error here will effectively disable that support

**Review by coroiu — CHANGES_REQUESTED** [review]
Hmm, are we sure that we

File: apps/browser/src/vault/fido2/content/page-script.ts (hunk 2 above)
Comment (Jingo88) [reply apps/browser/src/vault/fido2/content/page-script.ts]: Thanks for the suggestion. I moved the the logic into the try catch and updated the error to throw a `FallbackRequestedError`

File: apps/browser/src/vault/fido2/content/page-script.ts (hunk 1 above)
Comment (Jingo88) [reply apps/browser/src/vault/fido2/content/page-script.ts]: I tried shortening the logic to `options?mediation !== "optional` however some sides don't pass the `mediation` keyword at all, and in those cases I believe well want to keep the `options?.mediation` conditional and skip throwing the error.

File: apps/browser/src/vault/fido2/content/page-script.ts (hunk 2 above)
Comment (coroiu) [reply apps/browser/src/vault/fido2/content/page-script.ts]: Looks good!

**Review by coroiu — APPROVED** [review]
Looks good!

**Pull request merged**